From 0241e1c1b216d87050cdfcb4531ffc17f9eb1dc7 Mon Sep 17 00:00:00 2001
From: Thirunarayanan Balathandayuthapani
 <thirunarayanan.balathandayuth@oracle.com>
Date: Thu, 15 Dec 2016 15:38:06 +0530
Subject: [PATCH] Bug #24585978	INNODB: ASSERTION TOTAL_RECS > 0 FAILURE 
 	IN FILE DICT0STATS.CC

Analysis:
========
 There was missing bracket for IF conditon in dict_stats_analyze_index_level()
and it leads to wrong result.

Fix:
====
 Fix the IF condition in dict_stats_analyze_index_level() so that it satisfied
the if condtion only if level is zero.

Reviewed-by : Jimmy Yang <jimmy.yang@oracle.com>
---
 storage/innobase/dict/dict0stats.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- mariadb-10.1.orig/storage/innobase/dict/dict0stats.cc
+++ mariadb-10.1/storage/innobase/dict/dict0stats.cc
@@ -1168,10 +1168,10 @@ dict_stats_analyze_index_level(
 		leaf-level delete marks because delete marks on
 		non-leaf level do not make sense. */
 
-		if (level == 0 && srv_stats_include_delete_marked? 0:
+		if (level == 0 && (srv_stats_include_delete_marked ? 0:
 		    rec_get_deleted_flag(
 			    rec,
-			    page_is_comp(btr_pcur_get_page(&pcur)))) {
+			    page_is_comp(btr_pcur_get_page(&pcur))))) {
 
 			if (rec_is_last_on_page
 			    && !prev_rec_is_copied
